internal/services/network/web_application_firewall_policy_resource.go

```diff
@@ -791,8 +795,8 @@ func expandWebApplicationFirewallPolicyOverrideRules(input []interface{}) *[]net
 	return &results
 }
 
-func expandWebApplicationFirewallPolicyMatchCondition(input []interface{}) *[]network.MatchCondition {
-	results := make([]network.MatchCondition, 0)
+func expandWebApplicationFirewallPolicyMatchCondition(input []interface{}) []webapplicationfirewallpolicies.MatchCondition {
```

Comment:
(same here)

Reply:
this function will not return with an error, so do not need a pointer to slice?

Reply:
maybe the expand functions should return a pointer only when the SDK Field is a pointer too? or we always return a pointer no matter what the SDK is?